src/jupyterlabcontroller/services/lab.py

```diff
@@ -1015,16 +1028,28 @@ def in_progress(self) -> _LabOperation | None:
     async def cancel(self) -> None:
         """Cancel any existing operation.
 
-        This is called during process shutdown. All operations in progress are
-        cancelled, which may strand random lab state that we will detect and
-        clean up during reconciliation.
+        This is called during process shutdown and to abort a spawn in
```

Comment:
Wait a minute.  I may have just realized why it's mostly the bots that are exploding.

The JupyterHub UI doesn't let you delete a spawn that's in progress...but we might bypass that with mobu.

Reply:
mobu only interacts with JupyterHub, so if JupyterHub doesn't permit it, mobu can't do it.